### PR TITLE
(Feature) Design for add collateral  collateral modal

### LINKED ISSUE #266: Design for the "Add Collateral" / "Collateral modal"
**Design**:  

- Step 2's design needs an update to better accommodate the "Add Collateral" button.
- There's also a new modal window to add the collateral, and it needs a design too.

Both are already implemented.

**Some thoughts:**

- Maybe add the `+` button as a link (check the "Add funds" link in other sections).
- The details data in the modal can be really long sometimes, beware of that.
- Use the "Connect to wallet" modal (see below) as a starting point for the new modal.

**Step 2's current implementation**

**Modal's current implementation:**

**Connect to Wallet Modal:**

15 - Conditional Exchange - Select Wallet

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
Closes #266 

Depends on #272 being merged, do not merge before that.

Designs are https://zpl.io/aNn649n https://zpl.io/amwjBNQ https://zpl.io/bee6gpN https://zpl.io/anQ5o0Y

I had to refactor the "Connect to wallet" modal to add some consistency between the different modals on the app so, yeah, you'll see I've modified some files seemingly not related to this PR (but again: they are).

**Screenshots for "Connect to wallet" modal:**

Screen Shot 2019-12-19 at 14 52 27

**Screenshots for "Add Collateral" modal:**

Screen Shot 2019-12-19 at 15 46 22
Screen Shot 2019-12-19 at 15 46 44
Screen Shot 2019-12-19 at 15 47 22

### CODE REVIEW

**Comment by fvictorio** [comment]
Bikeshedding time: I like how in the designs the address becomes green when the address is validated. @gabitoesmiapodo do you think it's worth doing it?

**Comment by fvictorio** [comment]
Pushed a minor change: if the address had trailing whitespace it was being considered invalid.

**Comment by gabitoesmiapodo** [comment]
@fvictorio Yeah, I forgot about that. I'll add it.

**Comment by gabitoesmiapodo** [comment]
@fvictorio Done.

File: app/src/components/common/customizable_tokens_select/modal_collateral.tsx [hunk 1]
```diff
@@ -1,137 +1,107 @@
...
+              const { value } = event.target
+              setCollateralAddress(value.trim())
+            }}
+            placeholder="0xac5d29e67a53ee4903d59a4c929b718e1d575eee"
+            type="text"
+          />
+        }
+        error={errorMessage || ''}
+        title={'Collateral Token Address'}
+        tooltip={{ id: 'ERC20', description: 'Enter a valid ERC20 address.' }}
```
Comment (mariano-aguero) [inline app/src/components/common/customizable_tokens_select/modal_collateral.tsx]: Look really great! I have an small change, I think the text is not the same as in the designs, Can we update it?

Comment (gabitoesmiapodo) [reply app/src/components/common/customizable_tokens_select/modal_collateral.tsx]: I changed it on purpose, I like my version more. I find the original a little redundant.

**Pull request merged**